src/main/java/org/gridsuite/loadflow/server/service/LoadFlowService.java

```diff
@@ -0,0 +1,142 @@
+import org.gridsuite.loadflow.server.repositories.LoadFlowResultRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@Service
+public class LoadFlowService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoadFlowService.class);
+
+    @Value("${gridsuite.services.report-server.base-uri:http://report-server}/")
+    public String reportServerURI;
+
+    @Value("${loadflow.default-provider}")
+    private String defaultProvider;
+
+    private LoadFlowResultRepository resultRepository;
+
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    NotificationService notificationService;
+
+    private UuidGeneratorService uuidGeneratorService;
+
+    public LoadFlowService(NotificationService notificationService, LoadFlowResultRepository resultRepository, ObjectMapper objectMapper, UuidGeneratorService uuidGeneratorService) {
+        this.notificationService = Objects.requireNonNull(notificationService);
+        this.resultRepository = Objects.requireNonNull(resultRepository);
+        this.objectMapper = Objects.requireNonNull(objectMapper);
+        this.uuidGeneratorService = Objects.requireNonNull(uuidGeneratorService);
+    }
+
+    public static List<String> getProviders() {
+        return LoadFlowProvider.findAll().stream()
+                .map(LoadFlowProvider::getName)
+                .collect(Collectors.toList());
+    }
+
+    public String getDefaultProvider() {
+        return defaultProvider;
+    }
+
+    public void setStatus(List<UUID> resultUuids, String status) {
+        resultRepository.insertStatus(resultUuids, status);
+    }
+
+    public static Map<String, List<Parameter>> getSpecificLoadFlowParameters(String providerName) {
+        return LoadFlowProvider.findAll().stream()
+                .filter(provider -> providerName == null || provider.getName().equals(providerName))
+                .map(provider -> {
+                    List<Parameter> params = provider.getSpecificParameters().stream()
+                            .filter(p -> p.getScope() == ParameterScope.FUNCTIONAL)
+                            .collect(Collectors.toList());
+                    return Pair.of(provider.getName(), params);
+                }).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+    }
+
+    public UUID runAndSaveResult(LoadFlowRunContext runContext) {
+        Objects.requireNonNull(runContext);
+        UUID resultUuid = uuidGeneratorService.generate();
+
+        // update status to running status
+        setStatus(List.of(resultUuid), LoadFlowStatus.RUNNING.name());
```

Comment:
It should be in the `consumeRun` method no ?

Reply:
I don't see it as a function that could be in a context class

Reply:
I'm not sure to understand the answer.
In consumeRun we already set the status, ex l 185 
`resultRepository.insertStatus(List.of(resultContext.getResultUuid()), LoadFlowStatus.COMPLETED.name());`
In this method of LoadFlowService, you set the status RUNNING before it's actually running.
But I see it's already like that in other MS so I guess let it like that.

Reply:
Because once the user clicks the button to initiate the calculation on the front, we assume that the calculation has started. It doesn't matter if it is actually running on the back or not, from the front perspective, we consider it as being launched.

Reply:
Ok but we are in the back-end here... nevermind.